common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java

```diff
@@ -385,14 +389,26 @@ private static CommandResult executeCommand(@SuppressWarnings(WarningType.rawtyp
                         command.getParameters().getCorrelationId());
             }
         } else /* baseException == null */ {
-            if (result != null && result instanceof AcquireTokenResult) {
+            if (result instanceof AcquireTokenResult) {
                 //Handler handler, final BaseCommand command, BaseException baseException, AcquireTokenResult result
                 commandResult = getCommandResultFromTokenResult(baseException, (AcquireTokenResult) result,
                         command.getParameters().getCorrelationId());
             } else {
+
                 //For commands that don't return an AcquireTokenResult
                 commandResult = new CommandResult(CommandResult.ResultStatus.COMPLETED, result,
                         command.getParameters().getCorrelationId());
+
+                //If any record needs a refresh, those records need to be refreshed.
+                if(!((List<?>) result).isEmpty()
```

Comment:
OK, you probably want to check if it implements List before you cast.  Probably extract the variable to save linear space.  In general, though, can we give it a better type than List?

Reply:
It's just that at runtime, java doesn't really know the difference between List<?> and List.

Reply:
I see what you mean. I've added the changes.